pkg/executables/govc.go

```diff
@@ -618,12 +618,19 @@ func (g *Govc) NetworkExists(ctx context.Context, network string) (bool, error)
 	exists := false
 
 	err := g.retrier.Retry(func() error {
-		networkResponse, err := g.exec(ctx, "find", "-maxdepth=1", filepath.Dir(network), "-type", "n", "-name", filepath.Base(network))
+		networkResponse, err := g.exec(ctx, "find", "-maxdepth=1", "-json=true", filepath.Dir(network), "-type", "n", "-name", filepath.Base(network))
 		if err != nil {
 			return err
 		}
 
-		if networkResponse.String() == "" {
+		foundNetworks := make([]string, 0)
+		if err = json.Unmarshal(networkResponse.Bytes(), &foundNetworks); err != nil {
+			logger.V(2).Info(fmt.Sprintf("Failed unmarshalling govc response: %s, %v", networkResponse.String(), err))
+			exists = false
+			return nil
+		}
+
+		if len(foundNetworks) != 1 {
```

Comment:
nit: maybe we should return an error if `len(foundNetworks) > 1`?
since returning false in that case is a bit misleading

Reply:
yeah i would like to change the final user message in these cases from `network not found` to something like `multiple networks found`, but wasnt sure the best way to surface that difference.  Returning an error could serve that need?

Reply:
Yeah, I think that works for now
We should eventually change this method to something like `SearchNetwork(...) []string` and make the caller decide what's a valid result and what's not. But we can create a ticket for that.

Reply:
ok ill give that a shot